### DISCUSSION TITLE
specify a role to a activity

### DISCUSSION THREAD

**Discussion by tianweimol** [elsa-workflows/elsa-core#3270] [Q&A]
thanks for you answer
How to specify the execution role of a activity, eg. i have a activity what is approve by leader(Jone),when this activity is executed,only Jone can executed this  activity.Is there some demo i can look at?
thank you.

**Comment by sfmskywalker** [comment]
I’m afraid there aren’t any demos yet with this type of workflow where user roles are involved. But you can write custom activities that implement role checking. For example, one activity could be called **Approve By Leader** accepting a user ID or user role as a property. You can set this property from the designer, either as a literal value or using a JS expression. The activity then checks if the current user matches the assigned user or user role. The question is: how do you determine the current user when the workflow executes? Will you provide it as input when triggering the workflow?

**Reply by tianweimol** [reply]
thanks for you answer,you have answered this question in #3269.
thanks!
